Userland/Applications/Dictionary/Dictionary.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021, Robin Allen <r@foon.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/MappedFile.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+
+namespace Dictionary {
+
+class Dictionary {
+public:
+    Dictionary(char const* filename);
+
+    String definition_of(int word_index) const;
+
+    StringView word_at_index(int word_index) const;
+
+    int count_words() const { return m_num_words; }
+
+    void prefix_query(
+        String query,
+        int start_index,
+        int& first_index,
+        int& num_indices) const;
+
+private:
+    template<typename T>
+    struct Offset {
+        uint32_t value;
+    };
+
+    struct Definition {
+        char part_of_speech;
+        char text[];
+    } __attribute__((packed));
+
+    struct Sense {
+        Offset<Definition> definition_offset;
+    } __attribute__((packed));
+
+    struct WordData {
+        uint8_t num_senses;
```

Comment:
Conventionally we use `u8`, `u32` etc from AK/Types.h

Reply:
Done.